src/subcommand.rs

```diff
@@ -255,10 +255,22 @@ impl Subcommand {
     Ok(())
   }
 
-  fn format(config: &Config, ast: Ast, search: &Search) -> Result<(), Error<'static>> {
+  fn format(config: &Config, ast: Ast, src: &str, search: &Search) -> Result<(), Error<'static>> {
     config.require_unstable("The `--fmt` command is currently unstable.")?;
 
-    if let Err(io_error) =
+    if config.dry_run {
+      let new_src = ast.to_string();
+      if src != new_src {
+        use prettydiff::diff_words;
+        println!("Formatting diff:\n\n{}", diff_words(&src, &new_src));
```

Comment:
Since it's only used once, it's fine to not have a use statement at all:

```suggestion
        println!("Formatting diff:\n\n{}", prettydiff::diff_words(&src, &new_src));
```

Reply:
Done (but the new `similar` crate uses more symbols)